src/server/worker/pipeline/transform/registry.go

```diff
@@ -256,8 +266,8 @@ func (reg *registry) processJobStarting(pj *pendingJob) error {
 		return err
 	}
 	if len(failed) > 0 {
-		reason := fmt.Sprintf("inputs failed: %s", strings.Join(failed, ", "))
-		return reg.failJob(pj, reason)
+		reason := fmt.Sprintf("Cancelled because the following upstream pipelines failed: %s", strings.Join(failed, ", "))
```

Comment:
This just gives us the immediate upstream pipelines rather than the full path of not run / canceled jobs. This may be fine considering the job set will have the not run / canceled state for all of the jobs in the downstream path, but I just want to make sure this meets the requirements.

Reply:
Confirmed with @dgeorg42 that this suffices for the time being.